src/main/app/controller/courts/CourtTypesController.ts

```diff
@@ -0,0 +1,142 @@
+import autobind from 'autobind-decorator';
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import {CourtType, CourtTypeItem, CourtTypePageData} from '../../../types/CourtType';
+
+@autobind
+export class CourtTypesController {
+
+
+  emptyCourtCodeErrorMsg = 'Court code is required and must be numeric.';
+  getCourtTypesErrorMsg = 'A problem occurred when retrieving the list of court types.';
+  updateErrorMsg = 'A problem occurred when saving the court court types.';
+  emptyCourtTypesErrorMsg = 'One or more court types are required.';
+
+
+  public async get(
+    req: AuthedRequest,
+    res: Response,
+    updated = false,
+    error = '',
+    courtCourtTypes: CourtType[] = null): Promise<void> {
+    if (!courtCourtTypes) {
+
+      const slug: string = req.params.slug as string;
+      await req.scope.cradle.api.getCourtCourtTypes(slug)
+        .then((value: CourtType[]) => courtCourtTypes = value)
+        .catch(() => error += this.getCourtTypesErrorMsg);
+
+    }
+
+    let allCourtTypes: CourtType[] = [];
+
+    await req.scope.cradle.api.getCourtTypes()
+      .then((value: CourtType[]) => allCourtTypes = value)
+      .catch(() => error += this.getCourtTypesErrorMsg);
+
+
+    const pageData: CourtTypePageData = {
+      errorMsg: error,
+      updated: updated,
+      items: this.mapCourtTypeToCourtTypeItem(allCourtTypes, courtCourtTypes)
+
+    };
+
+    res.render('courts/tabs/typesContent', pageData);
+  }
+
+
+  public async put(req: AuthedRequest, res: Response): Promise<void> {
+
+    let courtCourtTypes: CourtType[] = null;
+
+    if (req.body.types) {
+
+      courtCourtTypes = this.mapBodyToCourtType(req.body);
+
+      if(courtCourtTypes.find( c => (c.name ==="Magistrates' Court" && !c.code ) || (c.name ==="Magistrates' Court" && isNaN(c.code))
```

Comment:
I wonder if we could reduce this by half by having a separate property or method that checks if a code is undefined, null or NaN and then using this in the if statement?

Reply:
Created private helper method to check for null and Nan entries